app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -1073,8 +1073,12 @@ class BrowserTabViewModel @Inject constructor(
     }
 
     private fun getUrlHeaders(url: String?): Map<String, String> {
-        url?.let {
-            return gpc.getHeaders(url)
+        url?.let { requestUrl ->
+            val customHeaders = mutableMapOf<String, String>()
+            customHeadersPlugin.getPlugins().forEach {
```

Comment:
should this be in a customHeadersProvider? All logic can be encapsulated there.

Reply:
Updated, thanks :+1: